dbms/src/Common/TargetSpecific.h

```diff
@@ -46,8 +46,32 @@
  * 4. TIFLASH_IMPLEMENT_MULTITARGET_FUNCTION: implement the function body
  */
 
+namespace DB
+{
+enum class TargetArch : UInt32
+{
+    Default = 0, /// Without any additional compiler options.
+    SSE4 = (1 << 0), /// SSE4
+    AVX = (1 << 1),
+    AVX2 = (1 << 2),
+    AVX512F = (1 << 3),
+    AVX512BW = (1 << 4),
+    AVX512VBMI = (1 << 5),
+    AVX512VBMI2 = (1 << 6),
+};
+
+/// Runtime detection.
+bool isArchSupported(TargetArch arch);
+
+String toString(TargetArch arch);
+} // namespace DB
```

Comment:
Seems duplicate with `libs/libcommon/include/common/simd.h`

Reply:
I think no? `libs/libcommon/include/common/simd.h` is a configuration that whether we ENABLE_AVX512 optimize.